### PR TITLE
added error for not solved, along with test

### PR DESCRIPTION
**Pull request by sbarratt**

### CODE REVIEW

**Comment by sbarratt** [comment]
PTAL @akshayka

File: diffcp/cone_program.py [hunk 1]
```diff
@@ -48,6 +48,10 @@ def solve_and_derivative_batch(As, bs, cs, cone_dicts, n_jobs=-1, warm_starts=No
     return pool.starmap(solve_and_derivative_wrapper, args)
 
 
+class SolverError(Exception):
```
Comment (akshayka) [inline diffcp/cone_program.py]: Expose this in `__init__.py`

File: tests.py [hunk 2]
```diff
@@ -311,6 +312,17 @@ def test_threading(self):
             np.testing.assert_allclose(results[i][1], results_thread[i][1])
             np.testing.assert_allclose(results[i][2], results_thread[i][2])
 
+    def test_infeasible(self):
+        try:
```
Comment (akshayka) [inline tests.py]: nit: use [assertRaises](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises)

```
c = np.ones(1)
b = np.array([1.0, -1.0])
A = sparse.csc_matrix(np.ones((2, 1)))
cone_dims = {"f": 2}
with self.assertRaises(cone_prog.SolverError, msg='Solver scs returned status.*'):
    cone_prog.solve_and_derivative(A, b, c, cone_dims)
```

File: diffcp/cone_program.py [hunk 3]
```diff
@@ -48,6 +48,10 @@ def solve_and_derivative_batch(As, bs, cs, cone_dicts, n_jobs=-1, warm_starts=No
...
+    pass
+
+
 def solve_and_derivative(A, b, c, cone_dict, warm_start=None, **kwargs):
     """Solves a cone program, returns its derivative as an abstract linear map.
```
Comment (akshayka) [inline diffcp/cone_program.py]: Add a Raises section to the bottom of the docstring

```

Returns:
    ...

Raises:
    SolverError: if the cone program is infeasible or unbounded
"""
```

File: diffcp/cone_program.py [hunk 4]
```diff
@@ -48,6 +48,10 @@ def solve_and_derivative_batch(As, bs, cs, cone_dicts, n_jobs=-1, warm_starts=No
     return pool.starmap(solve_and_derivative_wrapper, args)
 
 
+class SolverError(Exception):
+    pass
+
+
 def solve_and_derivative(A, b, c, cone_dict, warm_start=None, **kwargs):
```
Comment (akshayka) [inline diffcp/cone_program.py]: Document the behavior for infeasible/unbounded problems in the README, please? :)

File: tests.py [hunk 5]
```diff
@@ -311,6 +312,17 @@ def test_threading(self):
             np.testing.assert_allclose(results[i][1], results_thread[i][1])
             np.testing.assert_allclose(results[i][2], results_thread[i][2])
 
+    def test_infeasible(self):
```
Comment (akshayka) [inline tests.py]: nice, thanks

File: diffcp/cone_program.py (hunk 1 above)
Comment (sbarratt) [reply diffcp/cone_program.py]: done

File: tests.py (hunk 2 above)
Comment (sbarratt) [reply tests.py]: done

**Pull request merged**